bigquery_etl/cli/query.py

```diff
@@ -702,61 +717,54 @@ def backfill(
-                )
+        depends_on_past = metadata.scheduling.get("depends_on_past", False)
+        # If date_partition_parameter isn't set it's assumed to be submission_date:
+        # https://github.com/mozilla/telemetry-airflow/blob/dbc2782fa23a34ae8268e7788f9621089ac71def/utils/gcp.py#L194C48-L194C48
+        date_partition_parameter = metadata.scheduling.get(
+            "date_partition_parameter", "submission_date"
+        )
+        date_partition_offset = metadata.scheduling.get("date_partition_offset", 0)
 
-            match partitioning_type:
-                case PartitionType.DAY:
-                    dates = [
-                        start_date + timedelta(i)
-                        for i in range((end_date - start_date).days + 1)
-                    ]
-                case PartitionType.MONTH:
-                    dates = list(
-                        rrule(
-                            freq=MONTHLY,
-                            dtstart=start_date.replace(day=1),
-                            until=end_date,
-                        )
-                    )
-                    # Dates in excluded must be the first day of the month to match `dates`
-                    exclude = [
-                        date.fromisoformat(day).replace(day=1).strftime("%Y-%m-%d")
-                        for day in exclude
-                    ]
-                case _:
-                    raise ValueError(
-                        f"Unsupported partitioning type: {partitioning_type}"
+        # For backwards compatibility assume partitioning type is day
+        # in case metadata is missing
+        if metadata.bigquery:
+            partitioning_type = metadata.bigquery.time_partitioning.type
+        else:
+            partitioning_type = PartitionType.DAY
+            click.echo("Bigquery partitioning type not set. Using PartitionType.DAY")
+
+        match partitioning_type:
+            case PartitionType.DAY:
+                dates = [
+                    start_date + timedelta(i)
+                    for i in range((end_date - start_date).days + 1)
+                ]
+            case PartitionType.MONTH:
+                dates = list(
+                    rrule(
+                        freq=MONTHLY,
+                        dtstart=start_date.replace(day=1),
+                        until=end_date,
                     )
+                )
+                # Dates in excluded must be the first day of the month to match `dates`
+                exclude = [
+                    date.fromisoformat(day).replace(day=1).strftime("%Y-%m-%d")
+                    for day in exclude
+                ]
+            case _:
+                raise ValueError(f"Unsupported partitioning type: {partitioning_type}")
 
-        except FileNotFoundError:
-            click.echo(f"No metadata defined for {query_file_path}")
-
-        if depends_on_past and exclude != []:
+        if depends_on_past and exclude:
```

Comment:
The default is actually an empty `tuple` so this was being printed even if there's nothing in `exclude`